frontend/packages/wizard/src/confirm-action.tsx

```diff
@@ -0,0 +1,64 @@
+import React from "react";
+import {
+  Button,
+  ConfirmActionProps,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  TextField,
+  Typography,
+} from "@clutch-sh/core";
+
+interface ConfirmActionDialogProps extends ConfirmActionProps {
+  open: boolean;
+  onCancel: () => void;
+}
+
+const ConfirmAction: React.FC<ConfirmActionDialogProps> = ({
+  open,
+  title,
+  description,
+  confirmationText,
+  actionLabel,
+  onConfirm,
+  onCancel,
+}) => {
+  const [input, setInput] = React.useState("");
+
+  React.useEffect(() => setInput(""), [open]);
+
+  const handleConfirm = () => {
+    if (input === confirmationText) {
+      onConfirm();
+    }
```

Comment:
We'll probably want to leave this to the implementer as well. We will likely want a generic reason field versus something specific, better auditing that way.